### PR TITLE
vscode:added p.vsc.package option,p..vsc.extensions now can be used alongside imperatively installed extensions

### PR DESCRIPTION
**Pull request by pasqui23**

### CODE REVIEW

File: modules/programs/vscode.nix [hunk 1]
```diff
@@ -47,12 +57,16 @@ in
...
-    home.packages = [
-      (pkgs.vscode-with-extensions.override {
-        vscodeExtensions = cfg.extensions;
-      })
-    ];
+    home.packages = [ cfg.package ];
 
     home.file."${configFilePath}".text = builtins.toJSON cfg.userSettings;
+    #adapted from https://discourse.nixos.org/t/vscode-extensions-setup/1801/2
+    home.activation.vscExtensions = dag.entryAfter ["installPackages"] ''
```
Comment (rycee) [inline modules/programs/vscode.nix]: I don't see how this implementation removes an extension from `~/.vscode/extensions` when a user removes it from `programs.vscode.extensions`.

I think a better solution is to create a `home.file` entry for each extension.

File: modules/programs/vscode.nix [hunk 2]
```diff
@@ -18,6 +19,15 @@ in
   options = {
     programs.vscode = {
       enable = mkEnableOption "Visual Studio Code";
+      
+      package = mkOption {
+        type = types.package;
+        default = pkgs.vscode;
+        example = literalExample "pkgs.vscodium";
+        description = ''
+          Version of Visual Studio Code to install
```
Comment (rycee) [inline modules/programs/vscode.nix]: Descriptions should end in a period.

File: modules/programs/vscode.nix (hunk 1 above)
Comment (pasqui23) [reply modules/programs/vscode.nix]: How does one uses `home.file` to symlink a directory?

Comment (rycee) [reply modules/programs/vscode.nix]: Just set the `source` attribute to point to the directory. For example
```
home.file.test.source = ./this/is/a/directory;
```
should work fine.

**Comment by PierreR** [comment]
Does this PR solve the problem that any imperatively installed extension would fail with `cannot read property of undefined ` ?

The `shellcheck` extension when installed with the declarative approach does not seem to work (it is installed but disabled with no apparent way to enable it). That how I end up trying the imperative approach.

**Comment by pasqui23** [comment]
`shellcheck` is packaged in nixpkgs,but not as a vscode extension.

**Comment by PierreR** [comment]
@rycee Isn't this PR ready to be merged at one time or another ?

**Comment by rycee** [comment]
Hi! I'm travelling at the moment, I might have time to check on Sunday.

On 17 October 2019 23:32:31 CEST, Pi3r <notifications@github.com> wrote:
>@rycee Isn't this PR ready to be merged at one time or another ?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/rycee/home-manager/pull/795#issuecomment-543370765

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by rycee** [comment]
Squashed, added a few cleanups, and rebased to master in 024d1aa227978fe2dae2fb3e56bab9a7237c2401. Thanks!

**Pull request closed without merging**